pkg/apiserver/authentication/oauth/error.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oauth
+
+import "fmt"
+
+var (
+	// ErrorInvalidClient
```

Comment:
Isn't this code from somewhere?

Reply:
The following error type is defined in https://datatracker.ietf.org/doc/html/rfc6749#section-5.2

Reply:
Best to point out, include their license would be great

Reply:
Already added the external doc link.